Scripts/Runtime/Logging/.CSProject/UnityLogListener.cs

```diff
@@ -179,10 +200,19 @@ public void LogFormat(LogType logType, UnityEngine.Object context, string format
                 return;
             }
 
-            SendToLogger(context, LogTypeToLogLevel(logType), string.Format(format, args));
+            // Make UnityEngine.Debug.Assert failures throw an exception instead of just logging an error
+            StackTrace stackTrace = new StackTrace(fNeedFileInfo: true);
+            MethodBase assertMethod = stackTrace.GetFrame(2)?.GetMethod(); // Skip this function and UnityEngine.Logger
+            if (assertMethod != null && assertMethod.Name == "Assert" && assertMethod.DeclaringType.FullName == "UnityEngine.Debug")
+            {
+                throw new Exception(string.Format(format, args));
+            }
```

Comment:
This is open to debate, but given our decision that failed assertions should throw exceptions, I decided to intercept error logs from `UnityEngine.Debug.Assert` and convert them to exceptions. Let me know what you think.

Reply:
I think this makes sense to me. If something bad happens that  a Debug.Assert gets thrown, I really want to know about it and I think it should halt execution.

Reply:
yea, all good